### PR TITLE
Add anchors to headings with id

### PR DESCRIPTION
**Pull request by sbaughman**

### CODE REVIEW

File: add_heading_anchors.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+  init: function (selector) {
+    this.target = document.querySelector(selector);
+    if (this.target) {
+      this.addAnchorsToHeadings();
+    }
+  },
+
+  addAnchorsToHeadings: function () {
+    var headings = this.target.querySelectorAll('h1, h2, h3, h4, h5, h6'),
```
Comment (caffodian) [inline add_heading_anchors.js]: Wouldn't `h1[id] h2[id]` etc. be sufficient?  I get rusty with this sometimes

File: add_heading_anchors.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      id;
+
+    headings.forEach(function (heading) {
+      if (id = heading.id) {
```
Comment (caffodian) [inline add_heading_anchors.js]: This is supposed to be shorthand for 

```
if (heading.id) {
id = headingid;
....
}

```

right?  At first glance it's not obvious that it's not a typoed comparison.  This also may be redundant if the suggested selector works.

**Review by caffodian — CHANGES_REQUESTED** [review]
maybe can do some quick simplification by tweaking the selector.

File: add_heading_anchors.js (hunk 1 above)
Comment (sbaughman) [reply add_heading_anchors.js]: Yep good point.  Fixed!

File: add_heading_anchors.js (hunk 2 above)
Comment (sbaughman) [reply add_heading_anchors.js]: And yes, now redundant.  `if` check removed

**Review by caffodian — APPROVED** [review]

**Pull request merged**